### PR TITLE
[Docs] README revamp

### PR DESCRIPTION
**Pull request by MasterJH5574**
This PR revamps README.md for better clarity. Revamp of project landing page will follow up soon.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,62 +4,136 @@
 
 [Project Page](https://mlc.ai/mlc-llm/) | [Documentation](https://mlc.ai/mlc-llm/docs/) | [Blog](https://mlc.ai/blog/2023/05/01/bringing-accelerated-llm-to-consumer-hardware) | [WebLLM](https://webllm.mlc.ai/) | [WebStableDiffusion](https://websd.mlc.ai/) | [Discord][discord-url]
```
Comment (junrushao) [inline README.md:5]:
```suggestion
[Project Page](https://mlc.ai/mlc-llm/) | [Documentation](https://mlc.ai/mlc-llm/docs/) | [Blog](https://blog.mlc.ai/) | [WebStableDiffusion](https://websd.mlc.ai/) | [Discord][discord-url]
```

File: README.md [hunk 2]
```diff
@@ -4,62 +4,136 @@
...
-* AMD, Intel and NVIDIA GPUs via Vulkan on Windows and Linux;
-* NVIDIA GPUs via CUDA on Windows and Linux;
-* WebGPU on browsers (through companion project [WebLLM](https://github.com/mlc-ai/web-llm/tree/main)).
-
-<ins>**[Check out our instruction page to try out!](https://mlc.ai/mlc-llm/docs/get_started/try_out.html)**</ins>
-
-<p align="center">
-  <img src="site/gif/ios-demo.gif" height="700">
-</p>
+MLC LLM is a high-performance **universal deployment** solution for large-language models that allows **any language models** to be **deployed natively** on a diverse set of hardware backends and native applications, plus a **productive framework** for everyone to further optimize model performance for their own use cases.
```
Comment (junrushao) [inline README.md:7]:
```suggestion
Machine Learning Compilation for Large Language Models (MLC LLM) is a high-performance **universal deployment** solution that allows native deployment of any large language models with native APIs with compiler acceleration.
```

File: README.md [hunk 3]
```diff
@@ -4,62 +4,136 @@
...
+
+The cornerstone of MLC LLM is machine learning compilation (MLC) and the mission of MLC LLM is to **enable everyone to develop, optimize and deploy AI models natively on everyone's devices**.
```
Comment (junrushao) [inline README.md:9]:
```suggestion
The mission of the project is to enable everyone to develop, optimize and deploy AI models natively on everyone's devices with ML compilation techniques.
```

File: README.md [hunk 4]
```diff
@@ -4,62 +4,136 @@
...
-
-We heavily rely on open-source ecosystem, more specifically, [TVM Unity](https://discuss.tvm.apache.org/t/establish-tvm-unity-connection-a-technical-strategy/13344), an exciting latest development in the TVM project that enables python-first interactive MLC development experiences that allows us to easily compose new optimizations all in Python, and incrementally bring our app to the environment of interest. We also leveraged optimizations such as fused quantization kernels, first class dynamic shape support and diverse GPU backends.
-
-## Get Started with MLC-LLM
+<ins>**[Check out our instruction page to try out!](https://mlc.ai/mlc-llm/docs/get_started/try_out.html)**</ins>
 
-Please check our [documentation](https://mlc.ai/mlc-llm/docs/get_started/try_out.html) to start the journey with MLC-LLM.
+## Citation
+
+If you find MLC LLM useful in your work, please consider citing the project using the following format:
```
Comment (junrushao) [inline README.md:77]:
```suggestion
Please consider citing our project if you find it useful:
```

File: README.md [hunk 5]
```diff
@@ -4,62 +4,136 @@
...
+```bibtex
+@software{mlc-llm,
+    author = {MLC team},
+    title = {{MLC-LLM}},
+    url = {https://github.com/mlc-ai/mlc-llm},
+    year = {2023}
+}
+```
+
+The underlying compiler techniques employed by MLC LLM are outlined in the following papers:
```
Comment (junrushao) [inline README.md:88]:
```suggestion
The underlying compiler techniques of MLC LLM include:
```

File: README.md [hunk 6]
```diff
@@ -4,62 +4,136 @@
...
+</table>
+
+MLC LLM provides multiple sets of APIs across platforms and environments, which you can use to build
+your own LLM applications with native acceleration by MLC LLM. These include:
+* [Python API and gradio frontend](https://mlc.ai/mlc-llm/docs/deploy/python.html),
+* [Rest API with OpenAI API compatibility](https://mlc.ai/mlc-llm/docs/deploy/rest.html),
+* [C++ API](https://mlc.ai/mlc-llm/docs/deploy/cli.html),
+* [JavaScript API](https://mlc.ai/mlc-llm/docs/deploy/javascript.html) and [Web LLM](https://github.com/mlc-ai/web-llm),
+* [Swift API for iOS](https://mlc.ai/mlc-llm/docs/deploy/ios.html), and
+* [Android App](https://mlc.ai/mlc-llm/docs/deploy/android.html).
```
Comment (junrushao) [inline README.md:64]:
```suggestion

## Universal Deployment and APIs

* [Python API](https://mlc.ai/mlc-llm/docs/deploy/python.html);
* [Rest API with OpenAI API compatibility](https://mlc.ai/mlc-llm/docs/deploy/rest.html);
* [C++ API](https://mlc.ai/mlc-llm/docs/deploy/cli.html);
* [JavaScript API](https://mlc.ai/mlc-llm/docs/deploy/javascript.html) and [WebLLM](https://github.com/mlc-ai/web-llm);
* [Swift API and iOS App](https://mlc.ai/mlc-llm/docs/deploy/ios.html);
* [Java API and Android App](https://mlc.ai/mlc-llm/docs/deploy/android.html).
```

File: README.md [hunk 7]
```diff
@@ -4,62 +4,136 @@
...
 
 ## News
```
Comment (junrushao) [inline README.md:66]: Move "News" before "Universal Deployment APIs"

File: README.md [hunk 8]
```diff
@@ -4,62 +4,136 @@
...
+      address = {Carlsbad, CA},
+      pages = {578--594},
+      url = {https://www.usenix.org/conference/osdi18/presentation/chen},
+      publisher = {USENIX Association},
+      month = oct,
+  }
+  ```
+</details>
 
 ## Links
```
Comment (junrushao) [inline README.md:138]: Remove "WebLLM", "project pag", "iOS app" in "Links" section

File: README.md (hunk 1 above)
Comment (tqchen) [reply README.md:5]: let us remove project landing page link for now and instead first focus on the documentation. Then consider add it back later. The landing page will no longer host demos and will also contain info ptrs to docs

File: README.md (hunk 6 above)
Comment (MasterJH5574) [reply README.md:64]: I’m wondering if it is good to explicit mention that people can do secondary development with the API.

**Comment by MasterJH5574** [comment]
@junrushao Thanks for the comments! Updated and please take another look :-)

File: README.md (hunk 6 above)
Comment (junrushao) [reply README.md]: I do want to emphasize that our API surface is awesome awesome, but to be clear, those documents haven’t covered APIs yet, unfortunately. Let’s avoid overselling for now

File: README.md [hunk 9]
```diff
@@ -2,70 +2,143 @@
...
-<ins>**[Check out our instruction page to try out!](https://mlc.ai/mlc-llm/docs/get_started/try_out.html)**</ins>
-
-<p align="center">
-  <img src="site/gif/ios-demo.gif" height="700">
-</p>
+[Documentation](https://mlc.ai/mlc-llm/docs/) | [Blog](https://mlc.ai/blog/) | [WebLLM](https://webllm.mlc.ai/) | [WebStableDiffusion](https://websd.mlc.ai/) | [Discord][discord-url]
+
+Machine Learning Compilation for Large Language Models (MLC LLM) is a high-performance **universal deployment** solution that allows native deployment of any large language models with native APIs with compiler acceleration.
+
+The mission of this project is to enable everyone to develop, optimize and deploy AI models natively on everyone's devices with ML compilation techniques.
```
Comment (junrushao) [inline README.md:9]: Merge the two paragraphs

File: README.md [hunk 10]
```diff
@@ -2,70 +2,143 @@
...
 * [07/19/2023] Supports 7B/13B/70B Llama-2
 
-## What is MLC LLM?
-
-In recent years, there has been remarkable progress in generative artificial intelligence (AI) and large language models (LLMs), which are becoming increasingly prevalent. Thanks to open-source initiatives, it is now possible to develop personal AI assistants using open-sourced models. However, LLMs tend to be resource-intensive and computationally demanding. To create a scalable service, developers may need to rely on powerful clusters and expensive hardware to run model inference. Additionally, deploying LLMs presents several challenges, such as their ever-evolving model innovation, memory constraints, and the need for potential optimization techniques.
-
-The goal of this project is to enable the development, optimization, and deployment of AI models for inference across a range of devices, including not just server-class hardware, but also users' browsers, laptops, and mobile apps. To achieve this, we need to address the diverse nature of compute devices and deployment environments. Some of the key challenges include:
+## Universal Deployment APIs
```
Comment (junrushao) [inline README.md:62]: Move "Getting Started" before "Universal Deployment APIs"

File: README.md [hunk 11]
```diff
@@ -2,70 +2,143 @@
...
-- Deploying on the native environment of user devices, which may not have python or other necessary dependencies readily available.
-- Addressing memory constraints by carefully planning allocation and aggressively compressing model parameters.
+MLC LLM provides multiple sets of APIs across platforms and environments, which you can use to build
+your own LLM applications with native acceleration by MLC LLM. These include:
+* [Python API](https://mlc.ai/mlc-llm/docs/deploy/python.html),
+* [Rest API with OpenAI API compatibility](https://mlc.ai/mlc-llm/docs/deploy/rest.html),
+* [C++ API](https://mlc.ai/mlc-llm/docs/deploy/cli.html),
+* [JavaScript API](https://mlc.ai/mlc-llm/docs/deploy/javascript.html) and [Web LLM](https://github.com/mlc-ai/web-llm),
+* [Swift API for iOS App](https://mlc.ai/mlc-llm/docs/deploy/ios.html), and
+* [Java API and Android App](https://mlc.ai/mlc-llm/docs/deploy/android.html).
```
Comment (junrushao) [inline README.md:71]:
```suggestion
* [Python API](https://mlc.ai/mlc-llm/docs/deploy/python.html)
* [OpenAI-Compatible Rest API](https://mlc.ai/mlc-llm/docs/deploy/rest.html)
* [C++ API](https://mlc.ai/mlc-llm/docs/deploy/cli.html)
* [JavaScript API](https://mlc.ai/mlc-llm/docs/deploy/javascript.html) and [Web LLM](https://github.com/mlc-ai/web-llm)
* [Swift API for iOS App](https://mlc.ai/mlc-llm/docs/deploy/ios.html)
* [Java API and Android App](https://mlc.ai/mlc-llm/docs/deploy/android.html)
```

**Review by junrushao — APPROVED** [review]

**Pull request merged**